cpp/include/cuspatial/experimental/detail/algorithm/point_linestring_distance.cuh

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2021-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuspatial/experimental/geometry/linestring_ref.cuh>
+#include <cuspatial/experimental/geometry/segment.cuh>
+#include <cuspatial/vec_2d.hpp>
+
+namespace cuspatial {
+namespace detail {
+
+template <typename T>
+__device__ T proj2(segment<T> const& s, vec_2d<T> const& v)
+{
+  return dot(v - s.v1, s.v2 - s.v1);
+}
+
+template <typename T, typename LinestringRef>
+inline __device__ T point_linestring_distance(vec_2d<T> const& point,
```

Comment:
I wonder if we can simplify `point_to_segment_distance_squared` in `linestring.cuh` with this API?

Reply:
Yeah, I think it can be simplified, or factored in a way to maximize reuse.